processor/normalizesumsprocessor/processor.go

```diff
@@ -0,0 +1,336 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package normalizesumsprocessor
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+)
+
+type NormalizeSumsProcessor struct {
+	logger     *zap.Logger
+	transforms []Transform
+
+	history map[string]*startPoint
+}
+
+type startPoint struct {
+	dataType        pdata.MetricDataType
+	intDataPoint    *pdata.IntDataPoint
+	doubleDataPoint *pdata.DoubleDataPoint
+	lastIntValue    int64
+	lastDoubleValue float64
+}
+
+func newNormalizeSumsProcessor(logger *zap.Logger, transforms []Transform) *NormalizeSumsProcessor {
+	return &NormalizeSumsProcessor{
+		logger:     logger,
+		transforms: transforms,
+		history:    make(map[string]*startPoint),
+	}
+}
+
+// ProcessMetrics implements the MProcessor interface.
+func (nsp *NormalizeSumsProcessor) ProcessMetrics(ctx context.Context, metrics pdata.Metrics) (pdata.Metrics, error) {
+	var errors []error
+
+	for i := 0; i < metrics.ResourceMetrics().Len(); i++ {
+		rms := metrics.ResourceMetrics().At(i)
+		processingErrors := nsp.transformMetrics(rms)
+		errors = append(errors, processingErrors...)
+	}
+
+	if len(errors) > 0 {
+		return metrics, consumererror.Combine(errors)
+	}
+
+	return metrics, nil
+}
+
+func (nsp *NormalizeSumsProcessor) transformMetrics(rms pdata.ResourceMetrics) []error {
+	var errors []error
+
+	ilms := rms.InstrumentationLibraryMetrics()
+	for j := 0; j < ilms.Len(); j++ {
+		ilm := ilms.At(j).Metrics()
+		newSlice := pdata.NewMetricSlice()
+		for k := 0; k < ilm.Len(); k++ {
+			metric := ilm.At(k)
+			if nsp.shouldTransformMetric(metric) {
```

Comment:
Now that there's no renaming happening, do we need to have any configuration at all, or can we juts process any metric that doesn't have start times defined?

Reply:
There's currently two modes, one for do all sum metrics, one for targeting specific sum metrics.

There is no (to my understanding) "undefined" start time, are we comfortable stating that anything with a 0 start time (so epoch time, start of 1970) is undefined and would be modified by this? Seems like a reasonable assumption, just wanted to confirm.

Reply:
Yeah, that's a reasonable assumption.  This is proto3, so it defaults to 0, but also we can assume that we shouldn't be looking at programs that started in 1970.